sampler_test.go

```diff
@@ -164,28 +164,71 @@ func TestRateLimitingSampler(t *testing.T) {
 	assert.False(t, sampled)
 }
 
-func TestGuaranteedThroughputProbabilisticSamplerUpdate(t *testing.T) {
+func TestGuaranteedThroughputProbabilisticSampler_update(t *testing.T) {
 	samplingRate := 0.5
 	lowerBound := 2.0
-	sampler, err := NewGuaranteedThroughputProbabilisticSampler(lowerBound, samplingRate)
+	upperBound := 10.0
+	sampler, err := NewGuaranteedThroughputProbabilisticSampler(lowerBound, samplingRate, upperBound)
 	assert.NoError(t, err)
 
 	assert.Equal(t, lowerBound, sampler.lowerBound)
 	assert.Equal(t, samplingRate, sampler.samplingRate)
+	assert.Equal(t, upperBound, sampler.upperBound)
 
 	newSamplingRate := 0.6
 	newLowerBound := 1.0
-	err = sampler.update(newLowerBound, newSamplingRate)
+	newUpperBound := 20.0
+	err = sampler.update(newLowerBound, newSamplingRate, newUpperBound)
 	assert.NoError(t, err)
 
 	assert.Equal(t, newLowerBound, sampler.lowerBound)
 	assert.Equal(t, newSamplingRate, sampler.samplingRate)
+	assert.Equal(t, newUpperBound, sampler.upperBound)
 
 	newSamplingRate = 1.1
-	err = sampler.update(newLowerBound, newSamplingRate)
+	err = sampler.update(newLowerBound, newSamplingRate, newUpperBound)
 	assert.Error(t, err)
 }
 
+func TestGuaranteedThroughputProbabilisticSampler_upperBound(t *testing.T) {
+	lowerBound := 4.0
+	upperBound := 1.0
```

Comment:
doesn't make sense `lowerBound > upperBound`. Can we make a more realistic test?